pkg/monitoring/metrics/virt-controller/vmistats_collector_test.go

```diff
@@ -372,13 +372,40 @@ var _ = Describe("VMI Stats Collector", func() {
 			}
 		},
 			Entry("no interfaces", [][]string{}),
-			Entry("one interface", [][]string{{"default", "192.168.1.2", "InternalIP"}}),
+			Entry("one interface", [][]string{{"default", "", "192.168.1.2", "InternalIP"}}),
 			Entry("two interfaces", [][]string{
-				{"networkA", "170.170.170.170", "InternalIP"},
-				{"networkB", "180.180.180.180", "InternalIP"},
+				{"networkA", "", "170.170.170.170", "InternalIP"},
+				{"networkB", "", "180.180.180.180", "InternalIP"},
 			}),
 		)
 
+		It("should create metric for interfaces with empty name, but with interface name", func() {
+			vmi := &k6tv1.VirtualMachineInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "test-ns",
+					Name:      "testvmi",
+				},
+				Status: k6tv1.VirtualMachineInstanceStatus{
+					NodeName: "testNode",
+					Interfaces: []k6tv1.VirtualMachineInstanceNetworkInterface{
+						{
+							InfoSource:    "domain, multus-status",
+							InterfaceName: "net-0",
+						},
+						{
+							InfoSource:    "domain, multus-status",
+							InterfaceName: "net-1",
+						},
```

Comment:
`InterfaceName` is the interface name inside the guest itself and is reported only if the guest-agent is running there.

Therefore, `InfoSource` should also include `guest-agent` to reflect that.

I would also use `eth0` and `eth1` to reflect the interfaces names, but that is not a must.